src/App.tsx

```diff
@@ -5,16 +5,35 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [moviesFavoriteList, setMoviesFavoriteList] = useState<Movie[]>([]);
+
+  const addMovieToFavorites = (newMovie:Movie) => {
+    setMoviesFavoriteList((previousMoviesFavoriteList:Movie[]) => {
+      if (previousMoviesFavoriteList.some(
+        (movie:Movie) => movie.imdbId === newMovie.imdbId,
```

Comment:
```suggestion
        movie => movie.imdbId === newMovie.imdbId,
```
you don't need typing here, `previousMoviesFavoriteList` has `Movie[]` type